src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -6,1136 +6,200 @@ namespace Microsoft.Azure.ServiceBus
+            : this(Encoding.UTF8.GetBytes(body))
         {
-            get
-            {
-                // Body is consumed in 2 cases.
-                // - Someone called GetBody() - this happens in both send and receive.
-                // - Send operation "considered" a message been sent.
-                // Note: this is not a thread safe boolean. IsBodyConsumed == false
-                // can still lead to Send/GetBody() throwing exception saying body is consumed
-                // in a multi-threaded scenario.
-                this.ThrowIfDisposed();
-                return this.getBodyCalled == 1 || this.messageConsumed == 1;
-            }
         }
 
-        /// <summary>Specifies if message is a received message or not.</summary>
-        public bool IsReceived => this.receiverHeaders != null;
-
-        internal Guid LockTokenGuid
+        public Message(byte[] array)
+            : this(new ArraySegment<byte>(array))
         {
-            get
-            {
-                this.ThrowIfDisposed();
-                this.ThrowIfNotLocked();
-                return this.receiverHeaders.LockToken;
-            }
-
-            set
-            {
-                this.ThrowIfDisposed();
-                this.EnsureReceiverHeaders();
-
-                this.receiverHeaders.LockToken = value;
-                if (value != Guid.Empty)
-                {
-                    this.initializedMembers |= MessageMembers.LockToken;
-                }
-                else
-                {
-                    this.ClearInitializedMember(MessageMembers.LockToken);
-                }
-            }
         }
 
-        internal short PartitionId
+        public Message(ArraySegment<byte> arraySegment)
         {
-            get
-            {
-                this.ThrowIfDisposed();
-                return this.partitionId;
-            }
-
-            set
-            {
-                if (value < 0)
-                {
-                    throw Fx.Exception.AsError(new ArgumentOutOfRangeException(nameof(this.PartitionId)));
-                }
-
-                this.ThrowIfDisposed();
-                this.initializedMembers |= MessageMembers.PartitionId;
-                this.partitionId = value;
-            }
+            this.Body = arraySegment;
+            this.SystemProperties = new SystemPropertiesCollection();
+            this.UserProperties = new Dictionary<string, object>();
         }
 
-        /// <summary> Gets or sets the the Publisher. </summary>
-        /// <value> Identifies the Publisher Sending the Message. </value>
-        /// <exception cref="ObjectDisposedException">Thrown if message is in disposed state.</exception>
-        /// <exception cref="InvalidOperationException">Thrown if <seealso cref="PartitionKey"/> or <seealso cref="SessionId"/> are set to different values.</exception>
-        internal string Publisher
+        public ArraySegment<byte> Body { get; set; }
```

Comment:
So that makes sense in the context of the application that is handling the body as a source of multiple messages. For the general purpose, you could return `byte[]`. 

For the case where you need to operate on sections of the body it would be `new ArraySegment<byte>(msg.Body)`. Why to impose `ArraySegment<byte>` on all use cases?